### PR TITLE
Pirate guns

### PR DESCRIPTION
**Pull request by scaly-chimp**
## About the PR 
Adds some pirate themed weapons using sprites from CEV-Eris for mappers to use if they want to map a new pirate shuttle.

**Screenshots**
musket
icon
flintlock
icon
flaregun
icon
pirate revolver
icon

**Changelog**
No changelog because the guns need to be mapped in.

I'd also appreciate any balance/spritework feedback.

### CODE REVIEW

**Comment by mirrorcult** [comment]
linter failing

File: Resources/Textures/Objects/Weapons/Guns/Snipers/flintlock.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+      "name": "bolt-closed"
+    },
+    {
+      "name": "inhand-left",
+      "directions": 4
+    },
+    {
+      "name": "inhand-right",
+      "directions": 4
```
Comment (scaly-chimp) [inline Resources/Textures/Objects/Weapons/Guns/Snipers/flintlock.rsi/meta.json:28]: IIRC the inhands don't work for this gun and I have no clue why.

File: Resources/Textures/Objects/Weapons/Guns/Snipers/musket.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "y": 32
+  },
+  "states": [
+    {
+      "name": "inhand-right",
+      "directions": 4
+    },
+    {
+      "name": "inhand-left",
+      "directions": 4
```
Comment (scaly-chimp) [inline Resources/Textures/Objects/Weapons/Guns/Snipers/musket.rsi/meta.json:16]: same here

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Revolvers/revolvers.yml [hunk 3]
```diff
@@ -134,3 +134,32 @@
       path: /Audio/Weapons/Guns/MagOut/revolver_magout.ogg
     soundInsert:
       path: /Audio/Weapons/Guns/MagIn/revolver_magin.ogg
+
+- type: entity
+  name: Pirate revolver
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Revolvers/revolvers.yml:139]:
```suggestion
  name: pirate revolver
```

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml [hunk 4]
```diff
@@ -256,3 +256,50 @@
   - type: Appearance
     visuals:
     - type: BarrelBoltVisualizer
+
+- type: entity
+  name: flare gun
+  parent: LauncherBase
+  id: FlareGun
+  description: A compact, single-shot pistol that fires shotgun shells.
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml:264]: ..is that what a flare gun is

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml [hunk 5]
```diff
@@ -79,3 +79,56 @@
   - type: Appearance
     visuals:
     - type: BarrelBoltVisualizer
+
+- type: entity
+  name: Flintlock pistol
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:84]:
```suggestion
  name: flintlock pistol
```

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml [hunk 6]
```diff
@@ -79,3 +79,56 @@
...
+  parent: SniperBase
+  id: FlintlockPistol
+  description: A pirate's companion, Yarrr!
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:87]:
```suggestion
  description: A pirate's companion. Yarrr!
```

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml [hunk 7]
```diff
@@ -79,3 +79,56 @@
...
+      path: /Audio/Weapons/Guns/MagIn/bullet_insert.ogg
+  - type: Appearance
+    visuals:
+    - type: BarrelBoltVisualizer
+  - type: Item
+    icon: Objects/Weapons/Guns/Snipers/flintlock.rsi
+
+
+- type: entity
+  name: Musket
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:111]:
```suggestion
  name: musket
```

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml [hunk 8]
```diff
@@ -79,3 +79,56 @@
...
+  parent: SniperBase
+  id: Musket
+  description: This should've been in a museum long before you were born.
+  components:
+  - type: Sprite
+    sprite: Objects/Weapons/Guns/Snipers/musket.rsi
+  - type: RangedWeapon
+  - type: BoltActionBarrel
+    caliber: AntiMaterial
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:120]: i dont know if this makes any sense for muskets but Based

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml (hunk 4 above)
Comment (scaly-chimp) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml:264]: Yeah, you can actually load a flare gun with shotgun shells, but irl if the flare gun will probably explode if it's made out of plastic. You can probably find video on youtube.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml (hunk 8 above)
Comment (scaly-chimp) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:120]: I just thought making the muskets high damage would make them more unique, and while on paper it seems really powerful to have a 60 damage rifle/pistol I believe they are balanced by their inaccuracy, and the hassle of reloading after one shot in the midst of combat. Then again, I don't know much about balance so maybe it should be changed, it's up to you I guess.

**Pull request merged**